EventListener/ControllerListener.php

```diff
@@ -50,13 +50,14 @@ public function __construct(Reader $reader)
      */
     public function onKernelController(FilterControllerEvent $event)
     {
-        if (!is_array($controller = $event->getController())) {
+        if (!is_array($controller = $event->getController()) && !is_object($controller)) {
             return;
         }
 
-        $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($controller[0]) : get_class($controller[0]);
+        list($class, $methodName) = is_array($controller) ? $controller : array($controller, '__invoke');
```

Comment:
Ok. I reversed `object` and `class`, respectively for the controller object and the reflection class. Let me know if you think there is even better. Thanks.